enhancements/installer/kubernete-slb-owned-api-loadbalancers.md

```diff
@@ -0,0 +1,508 @@
+---
+title: kubernetes-slb-owned-api-loadbalancers
+authors:
+  - "@abhinavdahiya"
+reviewers:
+  - "@sttts"
+  - "@smarterclayton"
+approvers:
+  - "@sttts"
+  - "@smarterclayton"
+creation-date: 2020-05-06
+last-updated: 2020-08-26
+status: implementable
+see-also:
+  -   
+replaces:
+  - 
+superseded-by:
+  - 
+---
+
+# Kubernetes Service Owned API Loadbalancers
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+1. Service type Loadbalancer's ExternalTrafficPolicy should be Cluster or Local ?
+2. How to reserve private IPs on AWS ?
+
+## Summary
+
+Currently the installer is responsible for creating the loadbalancers for API and API-INT endpoints for the cluster. These loadbalancers 
+are only created at install-time and their lifecycle is not owned by the cluster. Since the Kubernetes API is hosted in pods in the cluster, 
+using the Kubernetes service type `Loadbalancer` to manage the lifecycle of the loadbalancers is next-step for self-hosted Kubernetes.
+
+## Motivation
+
+### Goals
+
+1. The API, API-INT loadbalancers are managed by Kubernetes service type `Loadbalancer`.
+2. Recovery of the cluster from some, to *all* of the control-plane DOWN scenarios should not be substantially more difficult. We do not want 
+   to trade cluster owned lifecycle for easy operations for harder recovery of control-plane.
+4. Support this deployment model for all platforms that support service type `Loadbalancer`.
+5. Allow customers to opt-out for user-provisioned workflows.
+
+### Non-Goals
+
+1. Updating existing clusters to new deployment strategy on upgrades.
+
+
+## Background
+
+Currently the bootstrapping happens in these three stages,
+
+**Stage 1:** Bootstrapping control plane on bootstrap host
+
+PUBLIC ZONE:api=public-lb-ip
+PRIVATE ZONE:api=private-lb-ip;api-int=private-lb-ip
+
+public-lb:bootstrap(green),master-0(red);master-1(red);master-2(red)
+private-lb:bootstrap(green);master-0(red);master-1(red);master-2(red)
+
+The `bootstrap` machine hosts the bootstrap-control-plane, and the kubelet on `master-{0,1,2}` use the private-lb to communicate with k8s API.
+
+**Stage 2:** Bootstrapping complete
+
+PUBLIC ZONE:api=public-lb-ip
+PRIVATE ZONE:api=private-lb-ip;api-int=private-lb-ip
+
+public-lb:bootstrap(red),master-0(green);master-1(green);master-2(green)
```

Comment:
afaik we go to step 2 on green,green,red,red. So the situation written here is much later.